core/src/main/scala/org/apache/spark/deploy/worker/ui/WorkerPage.scala

```diff
@@ -112,7 +112,15 @@ private[ui] class WorkerPage(parent: WorkerWebUI) extends WebUIPage("") {
       <td>
         <ul class="unstyled">
           <li><strong>ID:</strong> {executor.appId}</li>
-          <li><strong>Name:</strong> {executor.appDesc.name}</li>
+          <li><strong>Name:</strong>
+          {
+            if ({executor.state == ExecutorState.RUNNING}) {
+              <a href={executor.appDesc.appUiUrl}> {executor.appDesc.name}</a>
```

Comment:
This was originally set by:
```
    val webUrl = sc.ui.map(_.webUrl).getOrElse("")
```
The webUrl is only valid after bind(). But I think normally this should not be empty.